SmartDeviceLink-iOS/SmartDeviceLink/SDLLocalization.m

```diff
@@ -0,0 +1,198 @@
+// SDLLocalization.m
+//
+
+
+#import "SDLLocalization.h"
+
+/** Private definition extending setter for the properties. */
+@interface SDLLocalization ()
+
+@property (readwrite) NSString *language;
+
+@property (readwrite) NSString *region;
+
+@property (readwrite) NSBundle *defaultBundle;
+
+@property (readwrite) NSLocale *defaultLocale;
+
+@property (readwrite) NSBundle *fallbackBundle;
+
+@property (readwrite) NSLocale *fallbackLocale;
+
+@end
+
+@implementation SDLLocalization
+
+- (NSBundle *)bundle {
+    if ([self defaultBundle]) {
+        return [self defaultBundle];
+    } else if ([self fallbackBundle]) {
+        return [self fallbackBundle];
+    } else {
+        return nil;
+    }
+}
+
+- (NSLocale *)locale {
+    if ([self defaultLocale]) {
+        return [self defaultLocale];
+    } else if ([self fallbackLocale]) {
+        return [self fallbackLocale];
+    } else {
+        return nil;
+    }
+}
+
++ (instancetype)defaultLocalization {
+    static SDLLocalization *object = nil;
+    static dispatch_once_t token;
+
+    dispatch_once(&token, ^{
+        // get the locale identifier
+        NSString *localeIdentifier = [[NSLocale currentLocale] localeIdentifier];
```

Comment:
I looked at this because I know i had looked earlier at these capabilities. I'm quite certain that if I took my english based phone into japan, `currentLocale` would return Japanese.